lib/r10k/settings/container.rb

```diff
@@ -1,3 +1,4 @@
+require 'set'
```

Comment:
interesting, is that really required? I run the rspec tests on. Ruby 3.0 on arch linux and also use r10k and didnt spot any issues https://github.com/archlinux/svntogit-community/blob/packages/r10k/trunk/PKGBUILD

Reply:
It's definitely required on Ruby 3.1, it fails both locally and in the GHA without it.

Reply:
ahhh I mixed up ruby 3.0 and 3.1.

Reply:
I _think_ the plan is to go right to 3.1 for the next major release of puppet-agent, so that's what I tested with.